### PR TITLE
feat: Upstreaming rollkit-sdk

### PR DESCRIPTION
**Pull request by vuong177**
## Overview
Upstreaming rollkit-sdk

## Summary by CodeRabbit

- **New Features**
  - Introduced the RappDK development kit for creating applications using the Rollkit framework.
  - Added a new Sequencer module for managing validator sets in rollup operations.
  - Implemented scripts for initializing CometBFT nodes and running local development environments.
  - Added command-line tools for submitting upgrades and managing rollup processes.

- **Documentation**
  - Comprehensive guides for integrating RappDK into applications and migrating existing chains to Rollkit.
  - Updated README files and additional documentation for better onboarding and usability.
  - Detailed changes to the staking module's functionality within the Rollkit framework.

- **Bug Fixes**
  - Resolved issues with module parameter handling in the staking and sequencer modules.

- **Chores**
  - Added scripts for managing dependencies and automating proto code generation.

### CODE REVIEW

File: sdk/docs/instructions/integration.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    ...
+    )
+    ```
+
+    Use our fork rollkit version:
+
+
+    ```go
+    github.com/rollkit/rollkit => github.com/decentrio/rollkit v0.0.0-20240516071120-d40857416a55s
+    ```
```
Comment (Manav-Aggarwal) [inline sdk/docs/instructions/integration.md:27]: Do we need to use a forked rollkit version anymore? @vuong177 
Didn't we upstream the changes?

File: sdk/docs/intergrationv2.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ```
+
+    Use our fork rollkit version:
+
+
+    ```go
+    github.com/rollkit/rollkit => github.com/decentrio/rollkit v0.0.0-20240516071120-d40857416a55s
+    ```
+
+**Notice: Migration requires Rollkit to allow ABCI valset changes so using our fork version is for this. We're working with Rollkit team for upstream this feature ! [Issue Link](https://github.com/rollkit/rollkit/issues/1673) !**
```
Comment (Manav-Aggarwal) [inline sdk/docs/intergrationv2.md:24]: Same as above

File: sdk/docs/nodummytoken.md [hunk 3]
```diff
@@ -0,0 +1 @@
+// TODO
```
Comment (Manav-Aggarwal) [inline sdk/docs/nodummytoken.md:1]: What is this file used for?

**Review by Manav-Aggarwal — CHANGES_REQUESTED** [review]
Left some comments regarding updating the forked rollkit version to upstream

File: sdk/docs/instructions/integration.md (hunk 1 above)
Comment (vuong177) [reply sdk/docs/instructions/integration.md:27]: we can use mainline version now

**Review by Manav-Aggarwal — APPROVED** [review]

**Pull request merged**